src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1841,6 +1818,38 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
         }
 
         /// <summary>
+        /// Deletes the package directories for the supplied list of packages. If an error occurs (for example,
+        /// a directory is in use), marks the directory for deletion when VS is restarted.
+        /// </summary>
+        public async Task DeletePackageDirectoriesAsync(INuGetProjectContext nuGetProjectContext, HashSet<PackageIdentity> packages, CancellationToken token)
```

Comment:
Don't use `HashSet` on a public API, this could be an `IEnumerable` as far as the method is concerned